resharper/src/resharper-unity/Feature/Services/LiveTemplates/UnityProjectScopeProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Application;
+using JetBrains.ReSharper.Feature.Services.LiveTemplates.Context;
+using JetBrains.ReSharper.Feature.Services.LiveTemplates.Scope;
+using JetBrains.ReSharper.Plugins.Unity.Feature.Services.LiveTemplates.Scope;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Feature.Services.LiveTemplates
+{
+    // Provides the sope points that are valid for the given context
+    [ShellComponent]
+    public class UnityProjectScopeProvider : ScopeProvider
+    {
+        public UnityProjectScopeProvider()
+        {
+            // Used when creating scope point from settings
+            Creators.Add(TryToCreate<InUnityCSharpProject>);
+            Creators.Add(TryToCreate<InUnityCSharpAssetsFolder>);
+            Creators.Add(TryToCreate<InUnityCSharpEditorFolder>);
+            Creators.Add(TryToCreate<InUnityCSharpRuntimeFolder>);
+            Creators.Add(TryToCreate<InUnityCSharpFirstpassFolder>);
+            Creators.Add(TryToCreate<InUnityCSharpFirstpassEditorFolder>);
+            Creators.Add(TryToCreate<InUnityCSharpFirstpassRuntimeFolder>);
+        }
+
+        public override IEnumerable<ITemplateScopePoint> ProvideScopePoints(TemplateAcceptanceContext context)
+        {
+            var project = context.GetProject();
+            if (project == null)
+                yield break;
+
+            if (!project.IsUnityProject())
+                yield break;
+
+            // We could check for C# here, like InRazorCSharpProject, but we only really support C# Unity projects
+            // Are there any other types?
```

Comment:
Well, there could be pure UnityScript projects, but they are not supported by us right now and UnityScript is going to be turned off  soon anyway AFAIK.

Reply:
Also, you can have F# project using the plugin https://forum.unity.com/threads/f-kit.411420/

Reply:
Perhaps I shouldn't assume C# here, then